aicssegmentation/workflow/workflow_definition.py

```diff
@@ -7,9 +7,30 @@
 from aicssegmentation.util.directories import Directories
 from .workflow_step import WorkflowStep
 
+@dataclass
+class WorkflowDefinitionFromFile:
+    """
+        Definition of an aics-segmentation Workflow
+
+        This class only defines the workflow (i.e. the workflow characteristics and steps)
+        and is used either for building an executable Workflow object
+        or to access information about the Workflow without needing to execute it
+        """
+
+
+    name: str
+    steps: List[WorkflowStep]
+    from_file: bool
+
+    def __init__(self, name: str, steps: List[WorkflowStep]):
+        self.name = name
+        self.steps = steps
+        self.from_file = True
+
+
 
 @dataclass
-class WorkflowDefinition:
+class WorkflowDefinition(WorkflowDefinitionFromFile):
```

Comment:
We need to have `WorkflowDefinition` be the base class, so I would find some different naming here. The reason for that is that if the base class is a different, then we're going to have to refactor napari-aicssegmentation to reference this new class name. Not a huge deal but it's annoying.
The other reason is more by convention - usually the more "generic" name should be the base class, and you keep getting more specific as you go down the inheritance chain.

Reply:
We just need to find a name that makes sense for the prebuilt workflows. All the workflows are from a file so maybe something like `ClassicWorkflowDefinition` or `StandardWorkflowDefinition` or `PrebuiltWorkflowDefinition`. I hate all of them but 🤷

Reply:
@Aerendel  I'm going with `PrebuiltWorkflowDefinition` for the name. If I specify `WorkflowDefinition` for type- does that include the child class `PrebuiltWorkflowDefinition` ?

Reply:
No it doesn't. I kinda confused myself on this, my bad. This is still the proper way to design classes - but we'll have to do a tiny bit of type refactoring on the plugin wherever we use the extra features. No big deal.